zio-http/src/test/scala/zhttp/service/ServerSpec.scala

```diff
@@ -89,6 +89,17 @@ object ServerSpec extends HttpRunnableSpec(8088) {
           val res = app.requestBodyAsString()
           assertM(res)(equalTo("abc"))
         }
+      } + suite("Echo With getBody") {
+        val app = Http.collectM[Request] { case req @ Method.POST -> !! / "echoRequestBody" =>
+          for {
+            chunk <- req.getBody
+          } yield Response.text(chunk.map(_.toChar).mkString(""))
+        }
+        testM("POST Echo request body back") {
```

Comment:
Move the test to `def requestSpec`. You don't need such as complex application to test this failure.

Something like this should also produce the same error (You will need to check)
```scala
Http.collectM[Request] { case req => req.getBody.as(Response.ok) }
```

Reply:
Done

Reply:
I am still using POST and content, so as to keep close to the issue raised